pyccel/codegen/printing/fcode.py

```diff
@@ -2790,39 +2828,40 @@ def split_pos_code(line, endpos):
         result = []
         trailing = ' &'
         for line in lines:
-            if len(line)>72 and '!' in line[:72]:
+            if len(line) > 72 and '!' in line[:72]:
                 result.append(line)
-            elif len(line)>72 and ('"' in line[72:] or "'" in line[72:]):
-                pos=[]
-                quote = 0
-                double_quote = 0
-                mine = []
-                for idx, c in enumerate(line):
-                    if c == '\'':
-                        quote += 1
-                    if c == '\"':
-                        double_quote += 1
-                    if c == ' ' and quote%2 == 0 and double_quote %2 == 0 and idx > 72 and (not pos or idx > pos[-1] + 65):
-                        pos.append(idx)
+            elif len(line) > 72 and ('"' in line[72:] or "'" in line[72:]):
+                # finds all spaces outside double quotes
+                pattern = re.compile(' (?=(?:[^"]*"[^"]*")*[^"]*$)')
+                spaces_outside_dq = []
+                for match in pattern.finditer(line):
+                    spaces_outside_dq.append(match.start())
+                # finds all spaces outside quotes
+                pattern = re.compile(' (?=(?:[^\']*\'[^\']*\')*[^\']*$)')
+                spaces_outside_q = []
```

Comment:
Is there a way to run these two regex simultaneously? Also are these two cases the only ones where breaking a line at a white space would be problematic?

Reply:
I believe the syntax would be ` (?=(?:[^"]*"[^"]*")*[^"]*$)|(?=(?:[^\']*\'[^\']*\')*[^\']*$)`. You can use a [regex tester](https://regex101.com/) to check your regex (the one I linked to also includes an explanation of what each part of the regex is doing

Reply:
does not seem to work. I tested it [here](https://regex101.com/r/wV5bD0/1) with no luck. I think that the or cannot work here because it looks ahead and may undo a " with a '.

Reply:
I'm actually not sure how this regex got so complicated. `("[^"]*")|('[^']*')` seems to do the trick

Reply:
I'm not sure how this got so complicated. `("[^"]*")|('[^']*')` seems to do the trick. I would expect it to catch the gap between strings too, but in the regex tester this doesn't seem to be the case. If it does catch them then it is sufficient to check odd numbered groups

Reply:
Great, Thank you! i will refactor that part.